### PR TITLE
Run charmcraft-specific Github actions only from the canonical context

### PR DESCRIPTION
**Pull request by jardon**
Set CI to only run publish charm action if github repo owner is canonical.

### CODE REVIEW

**Comment by jardon** [comment]
This CI pipeline *should* allow for the linting, unit tests, and integration tests to be run before a merge and for the juju-specific jobs, `lib-check` and `publish-charm` to be run on a successful merge.  This allows PRs to pass CI without dealing with repo secrets and charm keyrings.  This PR also configures the jobs so that the publish job only occurs after a successful check-libs job enabling the publish job to only occur when all other jobs have been completed successfully.

**Comment by DomFleischmann** [comment]
@ca-scribner seems like @jardon wants to address precisely what we've been discussing lately, I'm not sure if I like the fact this is disabling the lib-check part. What are your thoughts?

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -9,9 +9,27 @@ on:
 
 jobs:
 
+  lib-check:
+    name: Check libraries
+    runs-on: ubuntu-latest
+    if: (github.repository_owner == 'canonical') && (github.event.pull_request.merged == true)
```
[the remark this answers is not in the document]
Comment (ca-scribner) [reply .github/workflows/publish.yaml:15]: Can you explain the reason behind also checking the `event.pull_request.merged` here?  I'm not sure what that protects against

**Comment by jardon** [comment]
> My advice was to ignore the failing lib and publish checks here (they're captured separately in [canonical/kubeflow-ci#1](https://github.com/canonical/kubeflow-ci/issues/1) and [canonical/kubeflow-ci#2](https://github.com/canonical/kubeflow-ci/issues/2)) and if everything else was good we'd merge this PR with the two failing checks.

My question @ca-scribner is that if the checks are going to fail consistently in this scenario, then why not skip them?  With the `github.event.pull_request.merged == true` the check will still be ran but only in the canonical context because of `github.repository_owner == 'canonical'`

**Comment by ca-scribner** [comment]
Yes, I agree we need to fix the CI to handle these checks properly (likely skip it if from a fork, run it if from a canonical repo, etc.).  But that's something to be discussed separately (the question of "does this change to the minio charm look good" feels independent of "does this change to the minio charm's CI look good").  Breaking it into two separate PRs so they can each be evaluated independently makes more sense to me.

File: .github/workflows/publish.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ca-scribner) [reply .github/workflows/publish.yaml:15]: but doesn't the `if: github.repository_owner == 'canonical'` handle the fork-context problem?  I think `github.event.pull_request.merged == true` will stop the check from running in some desired cases (I'm not sure what `github.event.pull_request.merged == true` actually means, I'd have to look it up.  But my guess is that it is not `true` in both `event == merge` and `event == pull_request`, and we run checks on both)

**Pull request closed without merging**